src/uct/tcp/tcp_cm.c

```diff
@@ -597,13 +597,15 @@ static ucs_status_t uct_tcp_cm_magic_number_handler_io_err_cb(void *arg,
 /* This function is called from async thread */
 static void uct_tcp_cm_magic_number_handler(int fd, void *arg)
 {
-    uct_tcp_ep_t *ep                    = arg;
-    uct_tcp_iface_t UCS_V_UNUSED *iface = ucs_derived_of(ep->super.super.iface,
-                                                         uct_tcp_iface_t);
+    uct_tcp_ep_t *ep       = arg;
+    uct_tcp_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                            uct_tcp_iface_t);
     ucs_status_t status;
     uint64_t magic_number;
 
-    status = ucs_async_remove_handler(fd, 0);
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
```

Comment:
are you sure we need UCS_ASYNC_BLOCK? 
ucs async operations are already thread-safe

Reply:
still not removed for uct_tcp_iface_listener_init